test/test.js

```diff
@@ -1397,6 +1402,13 @@ describe('Page', function() {
       expect(requests.length).toBe(5);
       expect(requests[2].resourceType()).toBe('document');
     });
+    it('should work with self requesting page', async({page, server}) => {
```

Comment:
Can you please drop this test? It doesn't cover anything new wrt to another one you've added here.

Reply:
Do you think so? I would like to hear more reasons.

With `page.setRequestInterception(true)` it goes different code path.
I thought that's the reason why we have such tests as redirect chain for both non interception and interception versions.

Reply:
> With page.setRequestInterception(true) it goes different code path.

@yujiosaka from the point of view of Page class and `page.goto` method specifically, there's no difference in the codepath. Or am I missing something?

Reply:
> Or am I missing something?

No! I understand your point, so I'll drop the test :)

Reply:
Done!
https://github.com/GoogleChrome/puppeteer/pull/1781/commits/83daf475a75dbe4344996a6a87be2e52269bccce